lib/settings.js

```diff
@@ -475,19 +475,50 @@ ${this.results.reduce((x, y) => {
     return restrictedRepos.filter((restrictedRepo) => { return RegExp(restrictedRepo).test(repoName) }).length > 0
   }
 
-  async eachRepositoryRepos(github, restrictedRepos, log) {
+  async eachRepositoryRepos (github, log) {
     log.debug('Fetching repositories')
-    return github.paginate('GET /installation/repositories').then(repositories => {
-      return Promise.all(repositories.map(repository => {
-        if (this.isRestricted(repository.name)) {
-          return null
-        }
 
-        const { owner, name } = repository
-        return this.updateRepos({ owner: owner.login, repo: name })
+    const processedRepos = new Set()
+    const results = []
+
+    // Process existing repositories
+    const existingRepoResults = await github.paginate('GET /installation/repositories')
+      .then(repositories => {
+        return Promise.all(repositories.map(repository => {
+          if (this.isRestricted(repository.name)) {
+            return null
+          }
+          const { owner, name } = repository
+          processedRepos.add(`${owner.login}/${name}`)
+          return this.updateRepos({ owner: owner.login, repo: name })
+        }))
       })
-      )
-    })
+
+    // Process missing repositories
+    const repoInConfigs = Object.values(this.repoConfigs)
+      .filter(config => config.repository?.name)
+      .map(config => {
+        return {
+          name: config.repository.name,
+          owner: config.repository.organization || this.repo.owner
+        }
+      })
+    const missingRepoResults = await Promise.all(
+      repoInConfigs
+        .filter(repo => {
+          return !processedRepos.has(`${repo.owner}/${repo.name}`) || this.isRestricted(repo.name)
```

Comment:
@PendaGTP this logic seems allow restricted repos because only time the `this.isRestricted` is checked only of a repo is processed. even the that logic seems to be incorrect.
```suggestion
          return !(processedRepos.has(`${repo.owner}/${repo.name}`) || this.isRestricted(repo.name))
          // This would be same as 
          // return !processedRepos.has(`${repo.owner}/${repo.name}`) && !this.isRestricted(repo.name)
```

Reply:
@decyjphr Thanks for your review. I've pushed an update that fixes this wrong logic.